httpx/client_test.go

```diff
@@ -53,29 +53,32 @@ func (mi *MetricsInstrument) Timer(name string, tags ...metrics.Tag) metrics.Tim
 	return mi.timerMock
 }
 
-type TimerMock struct {
+type TimerSpy struct {
 	metrics.Timer
-	TagsMap map[string]interface{}
-	Started bool
-	Stopped bool
+	TagMapMutex sync.RWMutex
```

Comment:
the mutex should be unexported and used ofnly inside the TimerSpy methods. 

usually is called `mux` and put above the fields that it protects, with an empty line above it:
```go
type TimerSpy struct {
    metrics.Timer

    mu sync.RWMutex
    TagsMap ...
}
```

Reply:
then I would have to create reading functions in order to make it concurrent?

Reply:
yes

Reply:
done